Zend/zend_builtin_functions.c

```diff
@@ -686,7 +678,7 @@ static void is_a_impl(INTERNAL_FUNCTION_PARAMETERS, zend_bool only_subclass) /*
 	} else if (Z_TYPE_P(obj) == IS_OBJECT) {
 		instance_ce = Z_OBJCE_P(obj);
 	} else {
-		RETURN_FALSE;
+		zend_argument_type_error(1, "must be of type object|string, %s given", zend_zval_type_name(obj));
```

Comment:
As the comment above indicates, this function has historically been used to test mixed values. The silent return false here is intentional.

(If we want to do any change here, it should be in the form of deprecation the function.)

Reply:
I removed this change